lib/keila/mailings/sender_adapters/adapter.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Keila.Mailings.SenderAdapters.Adapter do
+  @moduledoc """
+  Defines a sender adapter.
+
+  Sender adapters are used by Keila to support sending campaings through various services.
+
+  `schema_fields/0` define configuration fields which will dynamically be added to the `Keila.Mailing.Sender.Config` schema.
+
+  `changeset/2` will be used to build a changeset for the configuration fields before storing them in a `Keila.Mailing.Sender.Config` schema.
+
+  `to_swoosh_config/1` allows to retrieve configuration fields from the `Keila.Mailings.Sender.Config` and build a config that will be passed to `Swoosh`.
```

Comment:
I know I’m being super nitpicky now, but ideally the lines shouldn’t be much longer than 80 characters 😅

Reply:
No problem! I'm also im the 80 chars camp but I think I don't have it properly setup for elixir ind my editor.